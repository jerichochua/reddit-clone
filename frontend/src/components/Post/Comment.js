import React from 'react';
import dayjs from 'dayjs';
import relativeTime from 'dayjs/plugin/relativeTime';

dayjs.extend(relativeTime);

const Comment = ({ comment, username, onDelete }) => {
  const isAuthor = username === comment.author;

  return (
    <li className='comments-list-item'>
      <div className='comment-container'>
        <div className='comment-header'>
          <span className='comment-header-author'>{comment.author}</span>
          <span className='comment-header-timestamp'>
            {dayjs(comment.created_at).fromNow()}
          </span>
          {isAuthor && (
            <button
              value={comment.id}
              className='delete-button'
              onClick={onDelete}
            >
              Delete
            </button>
          )}
        </div>
        <div className='comment-body'>
          <p>{comment.content}</p>
        </div>
      </div>
    </li>
  );
};

export default Comment;
