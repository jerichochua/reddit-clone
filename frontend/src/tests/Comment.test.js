import { act } from 'react-dom/test-utils';
import renderer from 'react-test-renderer';
import Comment from '../components/Post/Comment';

const comment = {
  id: 1,
  author: 'test',
  content: 'test',
  created_at: '2020-01-01T00:00:00.000Z',
};

jest.mock('../util/getRelativeTime', () => () => 'timestamp');

it('renders correctly', () => {
  const tree = renderer
    .create(<Comment comment={comment} username='test' />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('opens edit form when edit button is clicked', () => {
  const tree = renderer.create(<Comment comment={comment} username='test' />);
  act(() => {
    tree.root.findByProps({ label: 'Edit' }).props.onClick();
  });
  expect(tree.toJSON()).toMatchSnapshot();
});

it('closes edit form when cancel button is clicked', () => {
  const tree = renderer.create(<Comment comment={comment} username='test' />);
  act(() => {
    tree.root.findByProps({ label: 'Edit' }).props.onClick();
  });
  act(() => {
    tree.root.findByProps({ label: 'Cancel' }).props.onClick();
  });
  expect(tree.toJSON()).toMatchSnapshot();
});
