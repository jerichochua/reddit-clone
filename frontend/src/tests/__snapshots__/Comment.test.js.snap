// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`closes edit form when cancel button is clicked 1`] = `
<li
  className="comments-list-item"
>
  <div
    className="comment-container"
  >
    <div
      className="comment-header"
    >
      <span
        className="comment-header-author"
      >
        test
      </span>
      <span
        className="comment-header-timestamp"
      >
        timestamp
      </span>
      <button
        className="post-button secondary"
        onClick={[Function]}
      >
        Edit
      </button>
      <button
        className="post-button danger"
        value={1}
      >
        Delete
      </button>
    </div>
    <div>
      <p
        className="comment-body"
      >
        test
      </p>
    </div>
  </div>
</li>
`;

exports[`opens edit form when edit button is clicked 1`] = `
<li
  className="comments-list-item"
>
  <div
    className="comment-container"
  >
    <div
      className="comment-header"
    >
      <span
        className="comment-header-author"
      >
        test
      </span>
      <span
        className="comment-header-timestamp"
      >
        timestamp
      </span>
      <button
        className="post-button secondary"
        onClick={[Function]}
      >
        Cancel
      </button>
      <button
        className="post-button danger"
        value={1}
      >
        Delete
      </button>
    </div>
    <div>
      <form>
        <textarea
          className="comment-edit-textarea"
          defaultValue="test"
          name="commentedit"
          required={true}
          rows="2"
        />
        <div
          className="comment-edit-footer"
        >
          <button
            className="form-btn"
            name="commenteditbtn"
            type="submit"
            value={1}
          >
            Edit
          </button>
        </div>
      </form>
    </div>
  </div>
</li>
`;

exports[`renders correctly 1`] = `
<li
  className="comments-list-item"
>
  <div
    className="comment-container"
  >
    <div
      className="comment-header"
    >
      <span
        className="comment-header-author"
      >
        test
      </span>
      <span
        className="comment-header-timestamp"
      >
        timestamp
      </span>
      <button
        className="post-button secondary"
        onClick={[Function]}
      >
        Edit
      </button>
      <button
        className="post-button danger"
        value={1}
      >
        Delete
      </button>
    </div>
    <div>
      <p
        className="comment-body"
      >
        test
      </p>
    </div>
  </div>
</li>
`;
